spaceshooter/spaceShooter.py

```diff
@@ -435,126 +435,150 @@ def update(self):
-    for hit in hits:
-        if hit.type == 'shield':
-            player.shield += random.randrange(10, 30)
-            if player.shield >= 100:
+            elif event.key == pygame.K_p:
+                paused = not paused
+    if paused == True:
+        draw_rect_alpha(screen, (0, 0, 0, 127), (0, 0, 800, 600))
+        draw_text(screen, "GAME PAUSED !!!", 60, WIDTH/2, (HEIGHT/2)-90)
+        draw_text(screen, "Press P to Resume", 40, WIDTH/2, HEIGHT/2)
+        draw_text(screen, "or ESC to Exit", 40, WIDTH/2, (HEIGHT/2)+40)
+        pygame.display.update()
+        continue
+    else:
+        # 2 Update
+        all_sprites.update()
+
+        hits = pygame.sprite.groupcollide(mobs, bullets, True, True)
+
+        for hit in hits:
+            score += 50 - hit.radius
+            expl = Explosion(hit.rect.center, 'lg')
+            all_sprites.add(expl)
+            if random.random() > 0.9:
+                pow = Pow(hit.rect.center)
+                all_sprites.add(pow)
+                powerups.add(pow)
+            newmob()
+
+        # check if the player collides with the mob
+        hits = pygame.sprite.spritecollide(
+            player, mobs, True, pygame.sprite.collide_circle)
+        for hit in hits:
+            player.shield -= hit.radius * 2
+            expl = Explosion(hit.rect.center, 'sm')
+            all_sprites.add(expl)
+            newmob()
+            if player.shield <= 0:
+                death_explosion = Explosion(player.rect.center, 'player')
+                all_sprites.add(death_explosion)
+                player.hide()
+                player.lives -= 1
                 player.shield = 100
-        if hit.type == 'gun':
-            player.powerup()
 
-    ## if player died and the explosion has finished, end game
-    if player.lives == 0 and not death_explosion.alive():
-        running = False
-        # menu_display = True
-        # pygame.display.update()
+        # if the player hit a power up
+        hits = pygame.sprite.spritecollide(player, powerups, True)
+        for hit in hits:
+            if hit.type == 'shield':
+                player.shield += random.randrange(10, 30)
+                if player.shield >= 100:
+                    player.shield = 100
+            if hit.type == 'gun':
+                player.powerup()
+
+        if player.lives == 0 and not death_explosion.alive():
+            # running = False
+            # menu_display = True
+            screen.fill(BLACK)
+            screen.blit(background, background_rect)
+            pygame.display.update()
+            while True:
+                ev = pygame.event.poll()
+                if ev.type == pygame.KEYDOWN:
+                    if ev.key == pygame.K_RETURN:
+                        break
+                    elif ev.key == pygame.K_ESCAPE:
+                        break
+                    elif ev.key == pygame.K_r:
+                        reset = True
+                        player.lives = 5
```

Comment:
Your commit message says 

> improved to 3

but I see 5. How is that?

Also, you haven't changed initial/default value in `Player.__init__`

https://github.com/tasdikrahman/spaceShooter/blob/1867d77d3b90fcddb78746208bfd2492d8fbed52/spaceshooter/spaceShooter.py#L169 

or made it match this one.